File: Napakalakiruby/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "cultist.rb"

class CultistPlayer<player
  
  attr_accessor:myCultistCard,:totalCultistPlayers
  
  def initialize(p,c)
    consCopia(p)
    @myCultistCard=c
    @@totalCultistPlayers+=1
  end
  
  def getCombatLevel
    plv=super
    percent=plv*20/100
    clv=myCultistCard.getGainedLevels*@@totalCultistPlayers
    lv=plv+percent+clv
    return lv
  end
  
  def getOponentLevel(m)
    return m.getCombatLevelAgainstCultistPlayer
  end
  
  def shouldConvert
    return false
  end
  
  def giveMeATreasure
    indice=rand(@visibleTreasures.length)
    return @visibleTreasures[indice]
  end
  
  def canYouGiveMeATreasure
     res=false
       if(@visibleTreasures.length>0)
         res=true
       end
       res
  end
end

File: Napakalaki/src/napakalaki/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

/**
 *
 * @author chen__000
 */
public class CultistPlayer extends Player {
    private static int totalCultistPlayers=0;
    private Cultist myCultistCard;
   

    public CultistPlayer(Player p,Cultist c) {
        super(p);
        this.myCultistCard=c;
        this.totalCultistPlayers++;
    }
    
    protected int getCombatLevel(){
        int nivel=super.getCombatLevel();
        int porcentaje=nivel*20/100;
        int cultislv=this.myCultistCard.getGainedLevels()*this.getTotalCultistPlayers();
        int lv=nivel+porcentaje+cultislv;
        return lv;
        
    }
    
    @Override
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    @Override
    protected boolean shouldConvert(){
        return false;
    }
    
    @Override
    protected Treasure giveMeATreasure(){
        int numero= (int) (Math.random()*this.visibleTreasures.size());
        return this.visibleTreasures.get(numero);
    }
    
    @Override
    protected boolean canYouGiveMeATreasure(){
         boolean res=false;
       if(this.enemy.visibleTreasures.size()>0){
           res=true;
       }
       return res;
    }
    
    public static int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
    
    
    
    
    
    
}
